StorageBagItem Client crash

Minecraft Version: 1.16.5

Forge Version: 36.0.43

Mod Version: 1.1.7

Single Player or Server: server

Describe problem (what you were doing; what happened; what should have happened):
dropped an item on the ground, had storage bag  set to pick up everything. and client crashed..
 


Log file link:


[02Mar2021 12:23:52.980] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer/]: CKY joined the game
[02Mar2021 12:23:53.021] [Server thread/DEBUG] [Silent Gear/]: Sending materials craftin item correction packet
[02Mar2021 12:23:53.022] [Server thread/DEBUG] [Silent Gear/]: Sending parts crafting item correction packet
[02Mar2021 12:23:53.022] [Server thread/DEBUG] [Silent Gear/]: Recalculating for CKY_'s Netherite Diamond Pickaxe
[02Mar2021 12:23:53.023] [Server thread/DEBUG] [Silent Gear/]:  - Durability: 2159.2998 -> 2159.2998 (+0.0) - mods: [1561, x1.3, +128, +2]
[02Mar2021 12:23:53.023] [Server thread/DEBUG] [Silent Gear/]:  - Enchantability: 16.0 -> 16.0 (+0.0) - mods: [10, x1.1, +5]
[02Mar2021 12:23:53.023] [Server thread/DEBUG] [Silent Gear/]:  - Rarity: 49.41926 -> 49.41926 (+0.0) - mods: [70, 1, +8]
[02Mar2021 12:23:53.023] [Server thread/DEBUG] [Silent Gear/]:  - Harvest Speed: 10.0 -> 10.0 (+0.0) - mods: [8, x1.13, +1]
[02Mar2021 12:23:53.023] [Server thread/DEBUG] [Silent Gear/]:  - Magic Damage: 1.3333334 -> 1.3333334 (+0.0) - mods: [1, x1.33]
[02Mar2021 12:23:53.023] [Server thread/DEBUG] [Silent Gear/]:  - Attack Damage: 6.0 -> 6.0 (+0.0) - mods: [3, x1.33, +1, +1]
[02Mar2021 12:23:53.023] [Server thread/DEBUG] [Silent Gear/]:  - Harvest Level: 4.0 -> 4.0 (+0.0) - mods: [↑3, ↑2, ↑4]
[02Mar2021 12:23:53.023] [Server thread/DEBUG] [Silent Gear/]:  - Attack Speed: -2.8 -> -2.8 (+0.0) - mods: [0, -2.8]
[02Mar2021 12:23:53.023] [Server thread/DEBUG] [Silent Gear/]:  - Repair Efficiency: 1.0 -> 1.0 (+0.0) - mods: [100%]
[02Mar2021 12:23:53.070] [Server thread/ERROR] [net.minecraftforge.eventbus.EventBus/EVENTBUS]: Exception caught during firing event: null
	Index: 12
	Listeners:
		0: NORMAL
		1: net.minecraftforge.eventbus.EventBus$$Lambda$2776/1796975861@5e1f2050
		2: ASM: class com.direwolf20.buildinggadgets.common.events.ItemPickupHandler GetDrops(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		3: ASM: class me.shedaniel.architectury.event.forge.EventHandlerImplCommon event(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		4: ASM: me.desht.pneumaticcraft.common.event.EventHandlerUniversalSensor@68bc01c8 onItemPickUp(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		5: ASM: stepsword.mahoutsukai.handlers.ServerHandler@e19c5e8 useItem(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		6: ASM: class mod.chiselsandbits.events.EntityItemPickupEventHandler pickupItems(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		7: ASM: wayoftime.bloodmagic.util.handler.event.WillHandler@17ac8a64 onItemPickup(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		8: ASM: Block{naturesaura:pickup_stopper} onPickup(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		9: net.minecraftforge.eventbus.EventBus$$Lambda$2776/1796975861@7ae8d443
		10: net.minecraftforge.eventbus.EventBus$$Lambda$2776/1796975861@539ebdce
		11: ASM: org.cyclops.cyclopscore.advancement.criterion.ModItemObtainedTrigger@57493ad7 onPickup(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		12: ASM: com.lothrazar.cyclic.event.ItemEvents@fae648e onPlayerPickup(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		13: ASM: vazkii.quark.content.tools.module.SeedPouchModule@29d01fa7 onItemPickup(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		14: LOWEST
		15: ASM: class com.hollingsworth.arsnouveau.common.event.EventHandler itemPickupEvent(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
java.lang.NullPointerException
	at com.lothrazar.cyclic.item.storagebag.StorageBagItem.tryInsert(StorageBagItem.java:234)
	at com.lothrazar.cyclic.event.ItemEvents.onPlayerPickup(ItemEvents.java:222)
	at net.minecraftforge.eventbus.ASMEventHandler_1757_ItemEvents_onPlayerPickup_EntityItemPickupEvent.invoke(.dynamic)
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85)
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302)
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283)
	at net.minecraftforge.event.ForgeEventFactory.onItemPickup(ForgeEventFactory.java:432)
	at net.minecraft.entity.item.ItemEntity.func_70100_b_(ItemEntity.java:321)
	at net.minecraft.entity.player.PlayerEntity.func_71044_o(PlayerEntity.java:546)
	at net.minecraft.entity.player.PlayerEntity.func_70636_d(PlayerEntity.java:517)
	at net.minecraft.entity.LivingEntity.func_70071_h_(LivingEntity.java:2158)
	at net.minecraft.entity.player.PlayerEntity.func_70071_h_(PlayerEntity.java:223)
	at net.minecraft.entity.player.ServerPlayerEntity.func_71127_g(ServerPlayerEntity.java:404)
	at net.minecraft.network.play.ServerPlayNetHandler.func_73660_a(ServerPlayNetHandler.java:207)
	at net.minecraft.network.NetworkManager.func_74428_b(NetworkManager.java:226)
	at net.minecraft.network.NetworkSystem.func_151269_c(NetworkSystem.java:134)
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:865)
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:291)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:787)
	at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:642)
	at net.minecraft.server.MinecraftServer.lambda$startServer$0(MinecraftServer.java:232)
	at java.lang.Thread.run(Thread.java:748)

[02Mar2021 12:23:53.072] [Server thread/FATAL] [net.minecraftforge.common.ForgeMod/]: Preparing crash report with UUID 2453035e-c2a8-4a7f-9129-c7c580c44f24
[02Mar2021 12:23:53.072] [Server thread/WARN] [net.minecraft.network.NetworkSystem/]: Failed to handle packet for 
net.minecraft.crash.ReportedException: Ticking player
	at net.minecraft.entity.player.ServerPlayerEntity.func_71127_g(ServerPlayerEntity.java:467) ~[?:?]
	at net.minecraft.network.play.ServerPlayNetHandler.func_73660_a(ServerPlayNetHandler.java:207) ~[?:?]
	at net.minecraft.network.NetworkManager.func_74428_b(NetworkManager.java:226) ~[?:?]
	at net.minecraft.network.NetworkSystem.func_151269_c(NetworkSystem.java:134) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:865) ~[?:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:291) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:787) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:642) ~[?:?]
	at net.minecraft.server.MinecraftServer.lambda$startServer$0(MinecraftServer.java:232) ~[?:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_282]
Caused by: java.lang.NullPointerException
	at com.lothrazar.cyclic.item.storagebag.StorageBagItem.tryInsert(StorageBagItem.java:234) ~[cyclic:1.16.5-1.1.7]
	at com.lothrazar.cyclic.event.ItemEvents.onPlayerPickup(ItemEvents.java:222) ~[cyclic:1.16.5-1.1.7]
	at net.minecraftforge.eventbus.ASMEventHandler_1757_ItemEvents_onPlayerPickup_EntityItemPickupEvent.invoke(.dynamic) ~[?:?]
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85) ~[eventbus-4.0.0.jar:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) ~[eventbus-4.0.0.jar:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) ~[eventbus-4.0.0.jar:?]
	at net.minecraftforge.event.ForgeEventFactory.onItemPickup(ForgeEventFactory.java:432) ~[forge:?]
	at net.minecraft.entity.item.ItemEntity.func_70100_b_(ItemEntity.java:321) ~[?:?]
	at net.minecraft.entity.player.PlayerEntity.func_71044_o(PlayerEntity.java:546) ~[?:?]
	at net.minecraft.entity.player.PlayerEntity.func_70636_d(PlayerEntity.java:517) ~[?:?]
	at net.minecraft.entity.LivingEntity.func_70071_h_(LivingEntity.java:2158) ~[?:?]
	at net.minecraft.entity.player.PlayerEntity.func_70071_h_(PlayerEntity.java:223) ~[?:?]
	at net.minecraft.entity.player.ServerPlayerEntity.func_71127_g(ServerPlayerEntity.java:404) ~[?:?]
	... 9 more
[02Mar2021 12:23:53.145] [Server thread/ERROR] [net.minecraftforge.eventbus.EventBus/EVENTBUS]: Exception caught during firing event: null
	Index: 12
	Listeners:
		0: NORMAL
		1: net.minecraftforge.eventbus.EventBus$$Lambda$2776/1796975861@5e1f2050
		2: ASM: class com.direwolf20.buildinggadgets.common.events.ItemPickupHandler GetDrops(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		3: ASM: class me.shedaniel.architectury.event.forge.EventHandlerImplCommon event(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		4: ASM: me.desht.pneumaticcraft.common.event.EventHandlerUniversalSensor@68bc01c8 onItemPickUp(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		5: ASM: stepsword.mahoutsukai.handlers.ServerHandler@e19c5e8 useItem(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		6: ASM: class mod.chiselsandbits.events.EntityItemPickupEventHandler pickupItems(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		7: ASM: wayoftime.bloodmagic.util.handler.event.WillHandler@17ac8a64 onItemPickup(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		8: ASM: Block{naturesaura:pickup_stopper} onPickup(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		9: net.minecraftforge.eventbus.EventBus$$Lambda$2776/1796975861@7ae8d443
		10: net.minecraftforge.eventbus.EventBus$$Lambda$2776/1796975861@539ebdce
		11: ASM: org.cyclops.cyclopscore.advancement.criterion.ModItemObtainedTrigger@57493ad7 onPickup(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		12: ASM: com.lothrazar.cyclic.event.ItemEvents@fae648e onPlayerPickup(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		13: ASM: vazkii.quark.content.tools.module.SeedPouchModule@29d01fa7 onItemPickup(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		14: LOWEST
		15: ASM: class com.hollingsworth.arsnouveau.common.event.EventHandler itemPickupEvent(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
java.lang.NullPointerException
	at com.lothrazar.cyclic.item.storagebag.StorageBagItem.tryInsert(StorageBagItem.java:234)
	at com.lothrazar.cyclic.event.ItemEvents.onPlayerPickup(ItemEvents.java:222)
	at net.minecraftforge.eventbus.ASMEventHandler_1757_ItemEvents_onPlayerPickup_EntityItemPickupEvent.invoke(.dynamic)
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85)
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302)
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283)
	at net.minecraftforge.event.ForgeEventFactory.onItemPickup(ForgeEventFactory.java:432)
	at net.minecraft.entity.item.ItemEntity.func_70100_b_(ItemEntity.java:321)
	at net.minecraft.entity.player.PlayerEntity.func_71044_o(PlayerEntity.java:546)
	at net.minecraft.entity.player.PlayerEntity.func_70636_d(PlayerEntity.java:517)
	at net.minecraft.entity.LivingEntity.func_70071_h_(LivingEntity.java:2158)
	at net.minecraft.entity.player.PlayerEntity.func_70071_h_(PlayerEntity.java:223)
	at net.minecraft.entity.player.ServerPlayerEntity.func_71127_g(ServerPlayerEntity.java:404)
	at net.minecraft.network.play.ServerPlayNetHandler.func_73660_a(ServerPlayNetHandler.java:207)
	at net.minecraft.network.NetworkManager.func_74428_b(NetworkManager.java:226)
	at net.minecraft.network.NetworkSystem.func_151269_c(NetworkSystem.java:134)
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:865)
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:291)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:787)
	at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:642)
	at net.minecraft.server.MinecraftServer.lambda$startServer$0(MinecraftServer.java:232)
	at java.lang.Thread.run(Thread.java:748)

[02Mar2021 12:23:53.147] [Server thread/FATAL] [net.minecraftforge.common.ForgeMod/]: Preparing crash report with UUID 03812c9c-4cbc-4c39-8116-a4a0b1cc7002
[02Mar2021 12:23:53.147] [Server thread/WARN] [net.minecraft.network.NetworkSystem/]: Failed to handle packet for 
net.minecraft.crash.ReportedException: Ticking player
	at net.minecraft.entity.player.ServerPlayerEntity.func_71127_g(ServerPlayerEntity.java:467) ~[?:?]
	at net.minecraft.network.play.ServerPlayNetHandler.func_73660_a(ServerPlayNetHandler.java:207) ~[?:?]
	at net.minecraft.network.NetworkManager.func_74428_b(NetworkManager.java:226) ~[?:?]
	at net.minecraft.network.NetworkSystem.func_151269_c(NetworkSystem.java:134) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:865) ~[?:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:291) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:787) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:642) ~[?:?]
	at net.minecraft.server.MinecraftServer.lambda$startServer$0(MinecraftServer.java:232) ~[?:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_282]
Caused by: java.lang.NullPointerException
	at com.lothrazar.cyclic.item.storagebag.StorageBagItem.tryInsert(StorageBagItem.java:234) ~[cyclic:1.16.5-1.1.7]
	at com.lothrazar.cyclic.event.ItemEvents.onPlayerPickup(ItemEvents.java:222) ~[cyclic:1.16.5-1.1.7]
	at net.minecraftforge.eventbus.ASMEventHandler_1757_ItemEvents_onPlayerPickup_EntityItemPickupEvent.invoke(.dynamic) ~[?:?]
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85) ~[eventbus-4.0.0.jar:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) ~[eventbus-4.0.0.jar:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) ~[eventbus-4.0.0.jar:?]
	at net.minecraftforge.event.ForgeEventFactory.onItemPickup(ForgeEventFactory.java:432) ~[forge:?]
	at net.minecraft.entity.item.ItemEntity.func_70100_b_(ItemEntity.java:321) ~[?:?]
	at net.minecraft.entity.player.PlayerEntity.func_71044_o(PlayerEntity.java:546) ~[?:?]
	at net.minecraft.entity.player.PlayerEntity.func_70636_d(PlayerEntity.java:517) ~[?:?]
	at net.minecraft.entity.LivingEntity.func_70071_h_(LivingEntity.java:2158) ~[?:?]
	at net.minecraft.entity.player.PlayerEntity.func_70071_h_(PlayerEntity.java:223) ~[?:?]
	at net.minecraft.entity.player.ServerPlayerEntity.func_71127_g(ServerPlayerEntity.java:404) ~[?:?]
	... 9 more
[02Mar2021 12:23:53.231] [Server thread/ERROR] [net.minecraftforge.eventbus.EventBus/EVENTBUS]: Exception caught during firing event: null
	Index: 12
	Listeners:
		0: NORMAL
		1: net.minecraftforge.eventbus.EventBus$$Lambda$2776/1796975861@5e1f2050
		2: ASM: class com.direwolf20.buildinggadgets.common.events.ItemPickupHandler GetDrops(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		3: ASM: class me.shedaniel.architectury.event.forge.EventHandlerImplCommon event(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		4: ASM: me.desht.pneumaticcraft.common.event.EventHandlerUniversalSensor@68bc01c8 onItemPickUp(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		5: ASM: stepsword.mahoutsukai.handlers.ServerHandler@e19c5e8 useItem(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		6: ASM: class mod.chiselsandbits.events.EntityItemPickupEventHandler pickupItems(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		7: ASM: wayoftime.bloodmagic.util.handler.event.WillHandler@17ac8a64 onItemPickup(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		8: ASM: Block{naturesaura:pickup_stopper} onPickup(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		9: net.minecraftforge.eventbus.EventBus$$Lambda$2776/1796975861@7ae8d443
		10: net.minecraftforge.eventbus.EventBus$$Lambda$2776/1796975861@539ebdce
		11: ASM: org.cyclops.cyclopscore.advancement.criterion.ModItemObtainedTrigger@57493ad7 onPickup(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		12: ASM: com.lothrazar.cyclic.event.ItemEvents@fae648e onPlayerPickup(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		13: ASM: vazkii.quark.content.tools.module.SeedPouchModule@29d01fa7 onItemPickup(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
		14: LOWEST
		15: ASM: class com.hollingsworth.arsnouveau.common.event.EventHandler itemPickupEvent(Lnet/minecraftforge/event/entity/player/EntityItemPickupEvent;)V
java.lang.NullPointerException
	at com.lothrazar.cyclic.item.storagebag.StorageBagItem.tryInsert(StorageBagItem.java:234)
	at com.lothrazar.cyclic.event.ItemEvents.onPlayerPickup(ItemEvents.java:222)
	at net.minecraftforge.eventbus.ASMEventHandler_1757_ItemEvents_onPlayerPickup_EntityItemPickupEvent.invoke(.dynamic)
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85)
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302)
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283)
	at net.minecraftforge.event.ForgeEventFactory.onItemPickup(ForgeEventFactory.java:432)
	at net.minecraft.entity.item.ItemEntity.func_70100_b_(ItemEntity.java:321)
	at net.minecraft.entity.player.PlayerEntity.func_71044_o(PlayerEntity.java:546)
	at net.minecraft.entity.player.PlayerEntity.func_70636_d(PlayerEntity.java:517)
	at net.minecraft.entity.LivingEntity.func_70071_h_(LivingEntity.java:2158)
	at net.minecraft.entity.player.PlayerEntity.func_70071_h_(PlayerEntity.java:223)
	at net.minecraft.entity.player.ServerPlayerEntity.func_71127_g(ServerPlayerEntity.java:404)
	at net.minecraft.network.play.ServerPlayNetHandler.func_73660_a(ServerPlayNetHandler.java:207)
	at net.minecraft.network.NetworkManager.func_74428_b(NetworkManager.java:226)
	at net.minecraft.network.NetworkSystem.func_151269_c(NetworkSystem.java:134)
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:865)
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:291)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:787)
	at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:642)
	at net.minecraft.server.MinecraftServer.lambda$startServer$0(MinecraftServer.java:232)
	at java.lang.Thread.run(Thread.java:748)


Video/images/gifs (direct upload or link):

--- a/src/main/java/com/lothrazar/cyclic/event/ItemEvents.java
+++ b/src/main/java/com/lothrazar/cyclic/event/ItemEvents.java
@@ -18,7 +18,6 @@ import com.lothrazar.cyclic.util.UtilChat;
 import com.lothrazar.cyclic.util.UtilItemStack;
 import com.lothrazar.cyclic.util.UtilSound;
 import com.lothrazar.cyclic.util.UtilWorld;
-import java.util.Set;
 import net.minecraft.block.BlockState;
 import net.minecraft.block.Blocks;
 import net.minecraft.entity.LivingEntity;
@@ -216,8 +215,7 @@ public class ItemEvents {
       ItemEntity itemEntity = event.getItem();
       ItemStack resultStack = itemEntity.getItem();
       int origCount = resultStack.getCount();
-      Set<Integer> bagSlots = StorageBagItem.getAllBagSlots(player);
-      for (Integer i : bagSlots) {
+      for (Integer i : StorageBagItem.getAllBagSlots(player)) {
         ItemStack bag = player.inventory.getStackInSlot(i);
         switch (StorageBagItem.getPickupMode(bag)) {
           case EVERYTHING:

--- a/src/main/java/com/lothrazar/cyclic/event/ItemEvents.java
+++ b/src/main/java/com/lothrazar/cyclic/event/ItemEvents.java
@@ -25,6 +25,7 @@ import net.minecraft.entity.LivingEntity;
 import net.minecraft.entity.ai.attributes.AttributeModifier;
 import net.minecraft.entity.ai.attributes.Attributes;
 import net.minecraft.entity.ai.attributes.ModifiableAttributeInstance;
+import net.minecraft.entity.item.ItemEntity;
 import net.minecraft.entity.player.PlayerEntity;
 import net.minecraft.item.ItemStack;
 import net.minecraft.potion.EffectInstance;
@@ -212,32 +213,29 @@ public class ItemEvents {
   public void onPlayerPickup(EntityItemPickupEvent event) {
     if (event.getEntityLiving() instanceof PlayerEntity) {
       PlayerEntity player = (PlayerEntity) event.getEntityLiving();
-      ItemStack stack = event.getItem().getItem();
-      ItemStack resultStack = null;
+      ItemEntity itemEntity = event.getItem();
+      ItemStack resultStack = itemEntity.getItem();
+      int origCount = resultStack.getCount();
       Set<Integer> bagSlots = StorageBagItem.getAllBagSlots(player);
       for (Integer i : bagSlots) {
         ItemStack bag = player.inventory.getStackInSlot(i);
         switch (StorageBagItem.getPickupMode(bag)) {
           case EVERYTHING:
             resultStack = StorageBagItem.tryInsert(bag, resultStack);
-            event.getItem().setItem(resultStack);
           break;
           case FILTER:
             resultStack = StorageBagItem.tryFilteredInsert(bag, resultStack);
-            event.getItem().setItem(resultStack);
           break;
           case NOTHING:
           break;
         }
-        if (resultStack == ItemStack.EMPTY) {
+        if (resultStack.isEmpty()) {
           break;
         }
       }
-      if (resultStack != null && resultStack.getCount() != stack.getCount()) {
+      if (resultStack.getCount() != origCount) {
+        itemEntity.setItem(resultStack);
         event.setResult(Result.ALLOW);
-      }
-      else {
-        event.setResult(Result.DEFAULT);
       }
     }
   }

--- a/src/main/java/com/lothrazar/cyclic/item/storagebag/StorageBagItem.java
+++ b/src/main/java/com/lothrazar/cyclic/item/storagebag/StorageBagItem.java
@@ -2,6 +2,7 @@ package com.lothrazar.cyclic.item.storagebag;
 
 import com.lothrazar.cyclic.base.ItemBase;
 import com.lothrazar.cyclic.registry.ContainerScreenRegistry;
+import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
@@ -310,8 +311,8 @@ public class StorageBagItem extends ItemBase {
     return RefillMode.NOTHING;
   }
 
-  public static Set<Integer> getAllBagSlots(PlayerEntity player) {
-    Set<Integer> slots = new HashSet<>();
+  public static List<Integer> getAllBagSlots(PlayerEntity player) {
+    List<Integer> slots = new ArrayList<>();
     for (int i = 0; i < player.inventory.getSizeInventory(); i++) {
       if (isBag(player.inventory.getStackInSlot(i))) {
         slots.add(i);

